[torch] Expose PCI info from CUDA device

efiks opened the pull request:
Summary:
PR#125083 add cuda device UUID info, but due to meta internal [version of ROCM the code was excluded](https://github.com/pytorch/pytorch/pull/125083?fbclid=IwY2xjawJvLnNleHRuA2FlbQIxMQABHlY55crrkTqWBWTsr2HVfuqnZ3R1GHR3o9Kf1o3h3uvyawEmCEdhdT48iY1P_aem_8tfrGrWE9SxFYasGfH8kCQ#issuecomment-2103315320).

This change will ensure meta internal code is built and PCI info is available

Test Plan: pass CI

Differential Revision: D73253426

Skylion007 commented on torch/testing/_internal/common_cuda.py:
```diff
@@ -106,6 +106,18 @@ def evaluate_platform_supports_fp8():
 
 PLATFORM_SUPPORTS_MX_GEMM: bool = LazyVal(lambda: TEST_CUDA and SM100OrLater)
 
+def evaluate_platform_supports_pci_info():
```
Please add return type bool

efiks commented:
@pytorchbot label "topic: not user facing"

Skylion007 commented on torch/testing/_internal/common_cuda.py:
```diff
@@ -106,6 +106,18 @@ def evaluate_platform_supports_fp8():
+    return False
+
+
+PLATFORM_SUPPORTS_PCI_INFO: bool = LazyVal(lambda: evaluate_platform_supports_pci_info())
```
```suggestion
PLATFORM_SUPPORTS_PCI_INFO: bool = LazyVal(evaluate_platform_supports_pci_info)
```
Useless lambda

eqy commented on torch/csrc/cuda/Module.cpp:
```diff
@@ -1085,8 +1085,11 @@ static void registerCudaDeviceProperties(PyObject* module) {
 #endif // USE_ROCM
           )
-#ifndef FBCODE_CAFFE2
+#if (defined(USE_ROCM) && ROCM_VERSION >= 60100) || !USE_ROCM
```
not sure if `!USE_ROCM` is the expressed intent here or rather just `defined(CUDA_VERSION)`

efiks replied:
This code shall be already compiled under cuda device. I've duplicated the line to evaluate if a right version of ROCm is used. In the past this code was failing to compiled within Meta, now with similar condition it does.

efiks committed:
[torch] Expose PCI info from CUDA device (#151672)

Summary:

PR#125083 add cuda device UUID info, but due to meta internal [version of ROCM the code was excluded](https://github.com/pytorch/pytorch/pull/125083?fbclid=IwY2xjawJvLnNleHRuA2FlbQIxMQABHlY55crrkTqWBWTsr2HVfuqnZ3R1GHR3o9Kf1o3h3uvyawEmCEdhdT48iY1P_aem_8tfrGrWE9SxFYasGfH8kCQ#issuecomment-2103315320).

This change will ensure meta internal code is built and PCI info is available

Test Plan: pass CI

Reviewed By: ngimel

Differential Revision: D73253426
